### PR TITLE
provider/azure: retry service principal creation

### PR DESCRIPTION
**Pull request by axw**
## Description of change

When running "juju add-credential azure", with the
interactive auth-type, retry creation of the service
principal and role assignment. This is necessary as
the AAD replication occurs asynchronously, and may
not be complete by the time we get to these steps.

## QA steps

juju add-credential azure
(interactive)

## Documentation changes

None.

## Bug reference

Fixes https://bugs.launchpad.net/juju/+bug/1680523

### CODE REVIEW

File: provider/azure/internal/azureauth/interactive.go [hunk 1]
```diff
@@ -206,17 +209,44 @@ func createOrUpdateServicePrincipal(
...
+				// The application doesn't exist yet, retry later.
+				return false
+			}
+			return true
 		},
-		nil, // abort
-	)
-	if err != nil {
+		Clock:       clock,
+		Delay:       time.Second,
```
Comment (wallyworld) [inline provider/azure/internal/azureauth/interactive.go]: Is this too short? Could DOS the API? Maybe every 2 or 3 seconds? What's the expected time to wait? Maybe tune to match that.

Comment (axw) [inline provider/azure/internal/azureauth/interactive.go]: I'll change to 2 seconds. Any longer and I think it'll be poor UX. We can fine tune based on feedback. I've never had to wait, but it all depends on how fast things happen, which is unpredictable.

Comment (axw) [inline provider/azure/internal/azureauth/interactive.go]: Actually I'll just make it 5 seconds, which is what azure-cli does. Should be fine.

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$argh$$

**Comment by axw** [comment]
$$merge$$

**Comment by axw** [comment]
$$mongoooooooo$$

**Pull request merged**